src/core/trtserver.cc

```diff
@@ -1646,6 +1681,46 @@ TRTSERVER_ServerModelRepositoryIndex(
   return nullptr;  // success
 }
 
+TRTSERVER_Error*
+TRTSERVER2_ServerModelRepositoryIndexNew(
+    TRTSERVER_Server* server, TRTSERVER2_ModelIndex** model_indices)
+{
+  ni::InferenceServer* lserver = reinterpret_cast<ni::InferenceServer*>(server);
+
+#ifdef TRTIS_ENABLE_STATS
+  ni::ServerStatTimerScoped timer(
+      lserver->StatusManager(), ni::ServerStatTimerScoped::Kind::REPOSITORY);
+#endif  // TRTIS_ENABLE_STATS
+
+  TrtServerModelIndex* indices = new TrtServerModelIndex();
+  *model_indices = reinterpret_cast<TRTSERVER2_ModelIndex*>(indices);
+  RETURN_IF_STATUS_ERROR(
+      lserver->GetModelRepositoryIndex(&indices->model_repository_index_));
```

Comment:
Don't expose model_repository_index_ publically. Just pass GetModelRepositoryIndex() into the constructor of TrtServerModelIndex.